### DISCUSSION TITLE
Buffer based samples question

### DISCUSSION THREAD

**Discussion by naushad-jamil** [danigb/smplr#87] [Q&A]
I saw an example of how to pass a buffer of sample sounds and play by name labels. That's great!

I have a couple questions:
1. What file formats are supported for the buffer? mp3 and anything else?
2. does the sampler have any feature to automatically generate pitches based on what pitches are given to it? Like if I provide C3 and G3 pitch can it generate the other pitches by computing?

**Comment by danigb** [comment]
Hi,

1. It allows all web audio api supported formats (ogg, mp3, wav...)
2. It does that internally, but is not easy to do it explicitly. That's something I want to add in next versions. What is you use case specifically?

**Reply by naushad-jamil** [reply]
Hi,

Thank you for responding. 

My use case is - I want to create an exotic instrument sound. I can record only a few pitches from that instrument. Say C4, G4. So I don't have the other pitches. How can I create a sampler instrument supplying these two pitches in one of the supported formats (say mp3) and have it play other pitches e.g. D4, E4, C5, C#5 etc.?

You said it can do it internally. Internally is good enough for my case.
